### PR TITLE
Issue 5375

### PR DESCRIPTION
**Pull request by easz**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

 - fix some existing unit test which was wrong
 - the validity of a property of a model will be checked in the constructor
 - add allowable enum values in python client code generation which can be further improved.
 - detail see #5375

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/python/model.mustache [hunk 1]
```diff
@@ -34,7 +41,15 @@ class {{classname}}(object):
         }
 
 {{#vars}}
-        self._{{name}} = {{name}}
+        self._{{name}} = None
+{{/vars}}
+
+        # TODO: let required properties as mandatory parameter in the constructor.
+        #       - to check if required property is not None (e.g. by calling setter)
+        #       - ApiClient.__deserialize_model has to be adapted as well
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/python/model.mustache]: @easz Thanks for the "TODO" note. 

Will we better off replacing ApiClient.__deserialize_model with [kim](https://github.com/mikeywaites/kim) or similar Python modules?

Comment (easz) [inline modules/swagger-codegen/src/main/resources/python/model.mustache]: I think the usual reflection technique is enough for that.

Comment (wing328) [inline modules/swagger-codegen/src/main/resources/python/model.mustache]: @easz thanks for the feedback. To be clear, I'm not saying we need to do it as part of this PR.  This is more for discussion on the future of the Python API client.

I've created https://github.com/swagger-api/swagger-codegen/issues/5421 for the discussion.

**Comment by wing328** [comment]
@easz thanks for the enhancements (including validity of a property of a model in the constructor).

**Pull request merged**